packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -85,21 +85,89 @@ class SystemChrome {
   /// embedder (if any). The embedder may choose to ignore unsupported
   /// overlays.
   ///
+  /// This method will schedule the embedder update to be run in a microtask.
+  /// Any subsequent calls to this method during the current event loop will
+  /// overwrite the pending value to be set on the embedder.
+  ///
   /// Arguments:
   ///
   ///  * [style]: A [SystemUiOverlayStyle] enum value denoting the style to use
   ///
   /// Return Value:
   ///
-  ///   boolean indicating if the preference was conveyed successfully to the
-  ///   embedder.
+  ///   The preference that was eventually conveyed to the embedder, along with
+  ///   whether it was successfully conveyed. This can be different than the
+  ///   value you specify here if another caller specified a different value
+  ///   later in the same event loop.
   ///
   /// Platform Specific Notes:
   ///
   ///   If the overlay is unsupported on the platform, enabling or disabling
   ///   that overlay is a no-op and always return true.
-  static Future<bool> setSystemUIOverlayStyle(SystemUiOverlayStyle style) async {
-    return (await _systemChromeProxy.setSystemUiOverlayStyle(
-      style)).success;
+  static Future<SystemUiOverlayStyleUpdate> setSystemUIOverlayStyle(SystemUiOverlayStyle style) {
+    assert(style != null);
+
+    if (_pendingStyleUpdate != null) {
+      _pendingStyleUpdate.style = style;
+      return _pendingStyleUpdate.future;
+    }
+
+    _pendingStyleUpdate = new _PendingStyleUpdate(style);
+    scheduleMicrotask(() {
+      assert(_pendingStyleUpdate != null);
+      if (_pendingStyleUpdate.style == _latestStyle) {
+        // No update needed; trivial success
+        _pendingStyleUpdate.complete();
+        _pendingStyleUpdate = null;
+        return;
+      }
+
+      _PendingStyleUpdate update = _pendingStyleUpdate;
+      _systemChromeProxy.setSystemUiOverlayStyle(update.style).then((dynamic value) {
```

Comment:
(i say this because on the next line you assume that the type is correct already, so you're already going to crash if it's wrong; better to make the code self-documenting and crash on the type check in checked mode)